prov/udp/src/udpx_ep.c

```diff
@@ -575,38 +575,30 @@ static int udpx_ep_close(struct fid *fid)
 	return 0;
 }
 
-static int udpx_ep_bind_cq(struct udpx_ep *ep, struct util_cq *cq, uint64_t flags)
+static int udpx_ep_bind_cq(struct udpx_ep *ep, struct util_cq *cq,
+			   uint64_t flags)
 {
 	struct util_wait_fd *wait;
 	int ret;
 
-	if (flags & ~(FI_TRANSMIT | FI_RECV)) {
-		FI_WARN(&udpx_prov, FI_LOG_EP_CTRL,
-			"unsupported flags\n");
-		return -FI_EBADFLAGS;
-	}
-
-	if (((flags & FI_TRANSMIT) && ep->util_ep.tx_cq) ||
-	    ((flags & FI_RECV) && ep->util_ep.rx_cq)) {
+	ret = ofi_ep_bind_cq(&ep->util_ep, cq, flags);
+	if (ret) {
 		FI_WARN(&udpx_prov, FI_LOG_EP_CTRL,
-			"duplicate CQ binding\n");
-		return -FI_EINVAL;
+			"CQ binding to EP is undable\n");
```

Comment:
See comment below.  We can drop the print here and move internal to bind, where we can report the reason for the failure.